File: src/main/scala/org/openapitools/client/api/Metadata.scala
package org.openapitools.client.api

import argonaut._
import argonaut.EncodeJson._
import argonaut.DecodeJson._

import org.http4s.{EntityDecoder, EntityEncoder}
import org.http4s.argonaut._
import org.joda.time.DateTime

import java.time.OffsetDateTime

import Metadata._

case class Metadata (
  /* HTTP status of the response (e.g., OK or ERROR) */
  status: Option[String],
/* HTTP method of the request */
  method: Option[String],
/* Requested URI path */
  `path`: Option[String],
/* Timestamp when response was generated */
  timestamp: Option[OffsetDateTime],
error: Option[ComponentsSchemasMetadataError])

object Metadata {
  import DateTimeCodecs._

  implicit val MetadataCodecJson: CodecJson[Metadata] = CodecJson.derive[Metadata]
  implicit val MetadataDecoder: EntityDecoder[Metadata] = jsonOf[Metadata]
  implicit val MetadataEncoder: EntityEncoder[Metadata] = jsonEncoderOf[Metadata]
}

File: client/javascript-closure-angular/API/Client/Metadata.js
goog.provide('API.Client.Metadata');

/**
 * @record
 */
API.Client.Metadata = function() {}

/**
 * HTTP status of the response (e.g., OK or ERROR)
 * @type {!string}
 * @export
 */
API.Client.Metadata.prototype.status;

/**
 * HTTP method of the request
 * @type {!string}
 * @export
 */
API.Client.Metadata.prototype.method;

/**
 * Requested URI path
 * @type {!string}
 * @export
 */
API.Client.Metadata.prototype.path;

/**
 * Timestamp when response was generated
 * @type {!Date}
 * @export
 */
API.Client.Metadata.prototype.timestamp;

/**
 * @type {!API.Client.components_schemas_Metadata_error}
 * @export
 */
API.Client.Metadata.prototype.error;

